zio/jsonio/writer.go

```diff
@@ -95,47 +95,50 @@ func (w *Writer) writeRecord(tab int, typ *zed.TypeRecord, bytes zcode.Bytes) {
 	tab += w.tab
 	w.punc('{')
 	it := bytes.Iter()
-	for i, f := range typ.Fields {
-		if i != 0 {
-			w.punc(',')
-		}
+	for _, f := range typ.Fields {
 		w.writeEntry(tab, f.Name, w.arena.New(f.Type, it.Next()))
+		if it.Done() {
+			w.newline()
+			w.indent(tab - w.tab)
+			break
+		}
+		w.punc(',')
 	}
-	w.newline()
-	w.indent(tab - w.tab)
 	w.punc('}')
 }
```

Comment:
This pattern's OK but I think what's going on here would be a lot clearer if it were conditioned on `len(bytes)`, as in

```go
	w.punc('{')
	if len(bytes) == 0 {
		w.punc('}')
		return
	}
	it := bytes.Iter()
	for _, f := range typ.Fields {
		if i != 0 {
			w.punc(',')
		}
		w.writeEntry(tab, f.Name, w.arena.New(f.Type, it.Next()))
	}
	w.newline()
	w.indent(tab - w.tab)
	w.punc('}')
```

or

```go
	w.punc('{')
	if len(bytes) > 0 {
		it := bytes.Iter()
		for _, f := range typ.Fields {
			if i != 0 {
				w.punc(',')
			}
			w.writeEntry(tab, f.Name, w.arena.New(f.Type, it.Next()))
		}
		w.newline()
		w.indent(tab - w.tab)
	}
	w.punc('}')
```